internal/core/project.go

```diff
@@ -148,6 +150,27 @@ func NewProject(ctx context.Context, os ...Option) (*Project, error) {
 		p.apps[appConfig.Name] = app
 	}
 
+	// configure pipelines for project and its apps
+	for _, name := range opts.Config.Pipelines() {
+		evalCtx := config.EvalContext(nil, p.dir.DataDir()).NewChild()
+		// TODO: Add variables
+
+		pipelineConfig, err := opts.Config.Pipeline(name, evalCtx)
+		if err != nil {
+			return nil, fmt.Errorf("error loading pipeline %q: %w", name, err)
+		}
+		if err := pipelineConfig.Validate(); err != nil {
```

Comment:
I need to double check but I think this will already happen! I am fairly sure now that we do this validation in `PipelinePut`

Reply:
OK yeah, we do this validation already when a pipeline is upserted, but not at the parsing step. I think it's worth it to get a config validation error since it's not supported storing tin the db so I'll update this PR for that!

Reply:
Handled in [d7391fd](https://github.com/hashicorp/waypoint/pull/3506/commits/d7391fde08a39605212327ffdc142ba89991efb2)!